src/client-side-metrics/gcp-metrics-handler.ts

```diff
@@ -0,0 +1,313 @@
+            unit: 'ms',
+            advice: {
+              explicitBucketBoundaries: latencyBuckets,
+            },
+          }
+        ),
+        firstResponseLatencies: meter.createHistogram(
+          'bigtable.googleapis.com/internal/client/first_response_latencies',
+          {
+            description:
+              'Latencies from when a client sends a request and receives the first row of the response.',
+            unit: 'ms',
+            advice: {
+              explicitBucketBoundaries: latencyBuckets,
+            },
+          }
+        ),
+        serverLatencies: meter.createHistogram(
+          'bigtable.googleapis.com/internal/client/server_latencies',
+          {
+            description:
+              'Latencies between the time when the Google frontend receives an RPC and when it sends the first byte of the response.',
+            unit: 'ms',
+
+            advice: {
+              explicitBucketBoundaries: latencyBuckets,
+            },
+          }
+        ),
+        connectivityErrorCount: meter.createCounter(
+          'bigtable.googleapis.com/internal/client/connectivity_error_count',
+          {
+            description:
+              "The number of requests that failed to reach Google's network. In normal cases, this number is 0. When the number is not 0, it can indicate connectivity issues between the application and the Google network.",
+          }
+        ),
+        clientBlockingLatencies: meter.createHistogram(
+          'bigtable.googleapis.com/internal/client/client_blocking_latencies',
+          {
+            description:
+              'Latencies introduced when the client blocks the sending of more requests to the server because of too many pending requests in a bulk operation.',
+            unit: 'ms',
+            advice: {
+              explicitBucketBoundaries: latencyBuckets,
+            },
+          }
+        ),
+      };
+    }
+  }
+
+  /**
+   * Records metrics for a completed Bigtable operation.
+   * This method records the operation latency and retry count, associating them with provided attributes.
+   * @param {OnOperationCompleteData} data Data related to the completed operation.
+   */
+  onOperationComplete(data: OnOperationCompleteData) {
+    this.initialize({
+      projectId: data.projectId,
+      instanceId: data.metricsCollectorData.instanceId,
+      table: data.metricsCollectorData.table,
+      cluster: data.metricsCollectorData.cluster,
+      zone: data.metricsCollectorData.zone,
+    });
+    this.otelMetrics?.operationLatencies.record(data.operationLatency, {
+      appProfileId: data.metricsCollectorData.appProfileId,
+      methodName: data.metricsCollectorData.methodName,
+      clientUid: data.metricsCollectorData.clientUid,
+      finalOperationStatus: data.finalOperationStatus,
+      streamingOperation: data.streamingOperation,
+      clientName: data.clientName,
```

Comment:
Shouldn't appProfileId, clientUid, and clientName be the same for all operations? If so, can't they be handle in `initialize`?

Reply:
> Shouldn't appProfileId, clientUid, and clientName be the same for all operations?

Yes, but in this case I chose to group them with attributes to be consistent with [requests in Golang](https://paste.googleplex.com/5938076009103360). Either way will work and either way has its tradeoffs. What do you think?

Reply:
> If so, can't they be handle in initialize

Yes, we could pass them into initialize and store them on the resource in this case.